plugins/runtime/threads/threads.service.ts

```diff
@@ -40,6 +39,29 @@ namespace Runtime {
     disableThreadContentionMonitoring(): angular.IPromise<void> {
       return this.jolokiaService.setAttribute('java.lang:type=Threading', 'ThreadContentionMonitoringEnabled', false);
     }
+    dumpThreads(): angular.IPromise<string> {
+      var dumpedThreads = "";
+      return this.jolokiaService.execute('java.lang:type=Threading', 'dumpAllThreads(boolean, boolean)', true, true)
+        .then(threads => {
+          threads.forEach(thread => {
+            thread.threadState = ThreadsService.STATE_LABELS[thread.threadState];
+            thread.waitedTime = thread.waitedTime > 0 ? Core.humanizeMilliseconds(thread.waitedTime) : '';
+            thread.blockedTime = thread.blockedTime > 0 ? Core.humanizeMilliseconds(thread.blockedTime) : '';
+          });
+          for (var i = 0; i < threads.length; i++) {
+            var tinfo = "";
+            var isdaemon = threads[i].daemon == true ? " daemon" : "";
+            tinfo += '"' + threads[i].threadName + '"' + " tid=" + threads[i].threadId.toString() + " prio=" + threads[i].priority.toString() + isdaemon + " java.lang.Thread.State:" + threads[i].threadState + "\n";
```

Comment:
why don't you use string interpolation here?

also, `tid` doesn't seem to be `tid` from jstack. Rather it should be `#nn` right after the thread name.